### PR TITLE
feat: git clone & lfs

### PR DESCRIPTION
**Pull request by TimPietrusky**
## Motivation

* Integrated GitPortable for Windows
* Use clone or lfs to interact with git
* Extended the installer to handle any amount of archives that need to be downloaded & extracted, so that git-portable doesn't need to be bundled with the installer

## Info

* Some e2e tests are only working on localhost, that's why the TEST_ENV `local` was added. Tests marked as "local" will be skipped in ci.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,7 +22,7 @@
     "test:client": "jest --runInBand --config jest.config.client.ts --verbose",
     "pretest:e2e": "nextron build --no-pack",
     "test:e2e": "xvfb-run --auto-servernum --server-args=\"-screen 0 1280x960x24\" -- npx playwright test",
```
Comment (pixelass) [inline package.json]:
```suggestion
    "test:e2e": "xvfb-run --auto-servernum --server-args=\"-screen 0 1280x960x24\" -- npx cross-env TEST_ENV=test npx playwright test",
```

File: playwright/git.test.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+let electronApp: ElectronApplication;
+let page: Page;
+
+test.beforeAll(async () => {
+	electronApp = await electron.launch({
+		args: ["."],
+		env: {
+			...process.env,
+			NODE_ENV: "development",
```
Comment (pixelass) [inline playwright/git.test.ts]:
```suggestion
```

File: src/electron/future/utils/git.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+		}
+	} else {
+		// Proceed with cloning
+		process = execa(git(), ["clone", "--progress", `git@hf.co:${repository}`, destinationPath]);
+
+		process.stderr?.on("data", (buffer: Buffer) => {
+			const output = buffer.toString();
+			const progressMatch = output.match(/Receiving objects:\s+(\d+)%/);
+			if (progressMatch) {
+				const percent = Number.parseFloat(progressMatch[1]);
```
Comment (pixelass) [inline src/electron/future/utils/git.ts]:
```suggestion
				const percent = Number.parseFloat(progressMatch[1]) / 100;
```

File: src/electron/future/utils/git.ts [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+	}
+
+	// Fetch LFS objects
+	process = execa(git(), ["lfs", "fetch"], { cwd: destinationPath });
+
+	process.stderr?.on("data", (buffer: Buffer) => {
+		const output = buffer.toString();
+		const progressMatch = output.match(/(\d+)% \((\d+)\/(\d+)\)/);
+		if (progressMatch) {
+			const percent = Number.parseFloat(progressMatch[1]);
```
Comment (pixelass) [inline src/electron/future/utils/git.ts]:
```suggestion
			const percent = Number.parseFloat(progressMatch[1]) / 100;
```

File: src/electron/future/utils/path-helpers.ts [hunk 5]
```diff
@@ -25,7 +26,13 @@ export function getDirectory(...subpath: string[]): string {
 }
 
 export function getUserData(...subpath: string[]): string {
-	return path.join(app.getPath("userData"), ...subpath);
+	const finalPath = path.join(app.getPath("userData"), ...subpath);
+
+	if (!existsSync(finalPath)) {
+		mkdirSync(finalPath, { recursive: true });
+	}
```
Comment (pixelass) [inline src/electron/future/utils/path-helpers.ts]: a getter should not create directories

**Review by pixelass — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (TimPietrusky) [reply package.json]: Thanks, done!

File: playwright/git.test.ts (hunk 2 above)
Comment (TimPietrusky) [reply playwright/git.test.ts]: Was removed.

File: src/electron/future/utils/path-helpers.ts (hunk 5 above)
Comment (TimPietrusky) [reply src/electron/future/utils/path-helpers.ts]: This was removed and a new helper was created.

**Review by pixelass — APPROVED** [review]
LGTM 🚀

**Pull request merged**